internal/util/ssa/patch.go

```diff
@@ -81,25 +92,63 @@ func Patch(ctx context.Context, c client.Client, fieldManager string, modified c
 		}
 	}
 
-	gvk, err := apiutil.GVKForObject(modified, c.Scheme())
+	gvk, err := apiutil.GVKForObject(modifiedUnstructured, c.Scheme())
 	if err != nil {
-		return errors.Wrapf(err, "failed to apply object: failed to get GroupVersionKind of modified object %s", klog.KObj(modified))
+		return errors.Wrapf(err, "failed to apply object: failed to get GroupVersionKind of modified object %s", klog.KObj(modifiedUnstructured))
 	}
 
 	patchOptions := []client.PatchOption{
 		client.ForceOwnership,
 		client.FieldOwner(fieldManager),
 	}
-	if err := c.Patch(ctx, modified, client.Apply, patchOptions...); err != nil {
-		return errors.Wrapf(err, "failed to apply %s %s", gvk.Kind, klog.KObj(modified))
+	if err := c.Patch(ctx, modifiedUnstructured, client.Apply, patchOptions...); err != nil {
+		return errors.Wrapf(err, "failed to apply %s %s", gvk.Kind, klog.KObj(modifiedUnstructured))
+	}
+
+	// Write back the modified object so callers can access the patched object.
+	if err := c.Scheme().Convert(modifiedUnstructured, modified, ctx); err != nil {
+		return errors.Wrapf(err, "failed to write modified object")
 	}
 
 	if options.WithCachingProxy {
 		// If the SSA call did not update the object, add the request to the cache.
-		if options.Original.GetResourceVersion() == modified.GetResourceVersion() {
+		if options.Original.GetResourceVersion() == modifiedUnstructured.GetResourceVersion() {
 			options.Cache.Add(requestIdentifier)
 		}
 	}
 
 	return nil
 }
+
+// prepareModified converts obj into an Unstructured and filters out undesired fields.
+func prepareModified(scheme *runtime.Scheme, obj client.Object) (*unstructured.Unstructured, error) {
+	u := &unstructured.Unstructured{}
+	switch obj.(type) {
+	case *unstructured.Unstructured:
+		u = obj.DeepCopyObject().(*unstructured.Unstructured)
+	default:
+		if err := scheme.Convert(obj, u, nil); err != nil {
+			return nil, errors.Wrap(err, "failed to convert object to Unstructured")
+		}
+	}
+
+	// Only keep the paths that we have opinions on.
+	FilterObject(u, &FilterObjectInput{
+		AllowedPaths: []contract.Path{
```

Comment:
This is the same as the default allowed paths in SSA + metadata.finalizers.

This should now cover all relevant fields